### DISCUSSION TITLE
Static server-side regeneration

### DISCUSSION THREAD

**Discussion by cmlbrnc** [vercel/next.js#14236] [Help] [answered]
I noticed that static generation creates every single page  as  html every time you build your project.
However,  It doesnt rerender the prerendered htmls pages on the serverless side. The only once is as the build time. So maybe database data updates, It doesnt show the updated data. I am asking If I have to use SSR generation or it is possible to rerender itself at the scope of SSG in that case. Thank you.

**Comment by rafaelalmeidatk** [comment, marked as answer]
You can use the experimental `unstable_revalidate` variable:

```js
export async function getStaticProps(context) {
  return {
    props: { ... },
    unstable_revalidate: 10, // if someone hits the page 10 seconds after the last build, the page will be regenerated
  }
}

**Reply by cmlbrnc** [reply]
Is it possible regenerate SSG when there is a difference  with SSR generation?

**Reply by rafaelalmeidatk** [reply]
Sorry I didn't understand the question, what do you mean by "difference with SSR generation"?. At the moment you can only regenerate static pages on the fly with the [`fallback`](https://nextjs.org/docs/basic-features/data-fetching#the-fallback-key-required) or the `unstable_revalidate` keys.

**Reply by cmlbrnc** [reply]
Okay. I got it now. I am going to implement your suggestions. Thank you.

**Reply by cmlbrnc** [reply]
Unfortunately,'unstable_revalidate' doesnt work for me on Lambda@edge. I think only option for me is SSR.  Although, It is very fast anyway.
